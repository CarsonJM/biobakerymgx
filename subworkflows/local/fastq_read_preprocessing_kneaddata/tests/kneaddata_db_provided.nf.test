nextflow_workflow {

    name "Test Subworkflow: FASTQ_READ_PREPROCESSING_KNEADDATA"
    script "../main.nf"
    workflow "FASTQ_READ_PREPROCESSING_KNEADDATA"

    tag "subworkflows"
    tag "subworkflows_local"
    tag "fastq_read_preprocessing_kneaddata"
    tag "fastq_read_preprocessing_kneaddata_db_provided"


    test(" kneaddata_db = !null & ['sarscov2']['illumina']['test_fastq_gz'] ") {

        when {
            params {
                load("$baseDir/tests/nf-test-local-db-params.json")
                outdir          = "$outputDir"
                max_cpus = 1
            }
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id: 'test' ],
                        [
                            file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true)
                        ]
                    ]
                )
                input[1] = Channel.of(
                    params.kneaddata_db
                )
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }

    test(" kneaddata_db = !null & ['sarscov2']['illumina']['test_fastq_gz'] - stub ") {

        options "-stub"

        when {
            params {
                load("$baseDir/tests/nf-test-local-db-params.json")
                outdir          = "$outputDir"
                max_cpus        = 1
            }
            workflow {
                """
                input[0] = Channel.of(
                    [
                        [ id: 'test' ],
                        [
                            file(params.test_data['sarscov2']['illumina']['test_1_fastq_gz'], checkIfExists: true),
                            file(params.test_data['sarscov2']['illumina']['test_2_fastq_gz'], checkIfExists: true)
                        ]
                    ]
                )
                input[1] = Channel.of(
                    params.kneaddata_db
                )
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert snapshot(workflow.out).match() }
            )
        }
    }
}
