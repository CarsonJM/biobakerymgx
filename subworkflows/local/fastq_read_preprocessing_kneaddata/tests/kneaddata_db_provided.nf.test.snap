{
    " kneaddata_db = !null & ['sarscov2']['illumina']['test_fastq_gz'] - stub ": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_kneaddata_paired_1.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_kneaddata_paired_2.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "1": [
                    "all_samples_kneaddata_read_counts.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "2": [
                    "versions.yml:md5,78fbce8ab53a0d6fb8dbf43cd7624846",
                    "versions.yml:md5,bb0ae5552fe7c380606c211bcf91f2c8"
                ],
                "preprocessed_fastq_gz": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_kneaddata_paired_1.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "test_kneaddata_paired_2.fastq.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "read_counts_tsv": [
                    "all_samples_kneaddata_read_counts.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                ],
                "versions": [
                    "versions.yml:md5,78fbce8ab53a0d6fb8dbf43cd7624846",
                    "versions.yml:md5,bb0ae5552fe7c380606c211bcf91f2c8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-15T11:30:35.029799432"
    },
    " kneaddata_db = !null & ['sarscov2']['illumina']['test_fastq_gz'] ": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_kneaddata_paired_1.fastq.gz:md5,acba94fdc94d1e4229beec12086a93af",
                            "test_kneaddata_paired_2.fastq.gz:md5,7c8a08782ad6705eced33d11d5032859"
                        ]
                    ]
                ],
                "1": [
                    "all_samples_kneaddata_read_counts.tsv:md5,542631f8eaae27f58bf2c7537d9325c8"
                ],
                "2": [
                    "versions.yml:md5,78fbce8ab53a0d6fb8dbf43cd7624846",
                    "versions.yml:md5,bb0ae5552fe7c380606c211bcf91f2c8"
                ],
                "preprocessed_fastq_gz": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "test_kneaddata_paired_1.fastq.gz:md5,acba94fdc94d1e4229beec12086a93af",
                            "test_kneaddata_paired_2.fastq.gz:md5,7c8a08782ad6705eced33d11d5032859"
                        ]
                    ]
                ],
                "read_counts_tsv": [
                    "all_samples_kneaddata_read_counts.tsv:md5,542631f8eaae27f58bf2c7537d9325c8"
                ],
                "versions": [
                    "versions.yml:md5,78fbce8ab53a0d6fb8dbf43cd7624846",
                    "versions.yml:md5,bb0ae5552fe7c380606c211bcf91f2c8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-02-15T11:30:23.292738609"
    }
}